### PR TITLE
prevents implanting fukken disks in boxes inside of people

### LINKED ISSUE #4347: It's possible to implant the NAD inside of someone.
**Problem Description**:
As long as the disk is in a box that's being implanted, it's completely possible to implant the NAD inside anyone willing.

Considering the disk can't be implanted normally, I'm pretty sure this is a pretty serious exploit. Knowledgeable players will completely shut down nukies this way.

**Steps to reproduce:**
- Put NAD inside a box
- Implant box
- Laugh in dchat as nukies try to cut the clown open

### PR DESCRIPTION
**Pull request by pinatacolada**
fixes #4347

### CODE REVIEW

File: code/modules/surgery/implant.dm [hunk 1]
```diff
@@ -175,6 +175,12 @@
 		to_chat(user, "<span class='warning'>Central command would kill you if you implanted the disk into someone.</span>")
 		return 0//fail
 
+	if(istype(tool, /obj/item/weapon/storage/box))
```
Comment (FlattestGuitar) [inline code/modules/surgery/implant.dm]: It's still possible to use fannypacks or wallets with this. I'd change it to just `/obj/item/weapon/storage`.

**Comment by PJB3005** [comment]
`locate()` is recursive, use that instead of all the other crap.

File: code/modules/surgery/implant.dm [hunk 2]
```diff
@@ -175,6 +175,12 @@
...
+		for(var/obj/item/I in tool.contents)
```
Comment (monster860) [inline code/modules/surgery/implant.dm]: Use `GetAllContents(tool)` instead. Or someone will put a wallet inside a box or something and put it in.

Comment (pinatacolada) [inline code/modules/surgery/implant.dm]: perfect timing *scream

File: code/modules/surgery/implant.dm [hunk 3]
```diff
@@ -175,6 +175,12 @@
 		to_chat(user, "<span class='warning'>Central command would kill you if you implanted the disk into someone.</span>")
 		return 0//fail
 
+	if(istype(tool, /obj/item/weapon/storage))
```
Comment (PJB3005) [inline code/modules/surgery/implant.dm]: You don't need this check here, at all.

File: code/modules/surgery/implant.dm (hunk 2 above)
Comment (pinatacolada) [inline code/modules/surgery/implant.dm]: eh, correct me if i'm wrong but locate() seems to do what i want already, I'd still need to iterate through what GetAllContents() would return

**Comment by pinatacolada** [comment]
on a side note, i noticed we can implant anything into people, backpacks, even shotguns, so that's fun

**Comment by fludd12** [comment]
Can't implant the nuke disk? That's okay, just implant the nuke instead.

**Pull request merged**